Client/FluentClientExtensions.cs

```diff
@@ -21,8 +21,8 @@ public static class FluentClientExtensions
         public static bool Remove<TFilter>(this List<IHttpFilter> filters)
             where TFilter : IHttpFilter
         {
-            TFilter filter = filters.OfType<TFilter>().FirstOrDefault();
-            return filter != null && filters.Remove(filter);
+            IEnumerable<TFilter> tFilters = filters.OfType<TFilter>();
+            return tFilters.Any() && filters.Remove(tFilters.First());
```

Comment:
And for the record, I want to point out that my code does NOT search filters thrice. It searches once, the 'Any()' call will search the result will stop searching as soon as it finds one item, and the 'First()' will simply return the first item from the result without having to search through all items.